gnucash/gnome/dialog-print-check.c

```diff
@@ -2112,7 +2117,39 @@ draw_page_items(GtkPrintContext *context,
     trans = xaccSplitGetParent(pcd->split);
     /* This was valid when the check printing dialog was instantiated. */
     g_return_if_fail(trans);
-    amount = gnc_numeric_abs(xaccSplitGetAmount(pcd->split));
+
+    if (pcd->account)
+    {
+        /* A parent account, e.g. from a subaccount register plugin page.
+         * Subtotal the amount of all splits from children accounts. */
+        GList *children_accounts = NULL;
+        GList *node, *child;
+        children_accounts = gnc_account_get_descendants(pcd->account);
+        children_accounts = g_list_append(children_accounts, pcd->account);
+        amount = gnc_numeric_zero();
+
+        for (node = xaccTransGetSplitList(trans); node; node = node->next)
+        {
+            Split *split = node->data;
+            Account* acct = xaccSplitGetAccount(split);
+            for (child = children_accounts; child; child = child->next)
+            {
+                if (acct == child->data)
+                {
+                    amount = gnc_numeric_add_fixed(amount, xaccSplitGetAmount(split));
+                    break;
+                }
+            }
+        }
+        g_list_free(children_accounts);
+        children_accounts = NULL;
+    }
+    else
+    {
+        /* Print just the amount of the split. */
+        amount = xaccSplitGetAmount(pcd->split);
+    }
+    amount = gnc_numeric_abs(amount);
```

Comment:
The computation should be a separate function so that `draw_page_item` calls `amount = compute_amount(pcd);`.